### PR TITLE
PV63: GetNodeData

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: libethereum/EthereumPeer.cpp [hunk 1]
```diff
@@ -474,6 +474,40 @@ bool EthereumPeer::interpret(unsigned _id, RLP const& _r)
...
+		unsigned count = static_cast<unsigned>(_r.itemCount());
+		if (!count)
+		{
+			clog(NetImpolite) << "Zero-entry GetNodeData: Not replying.";
+			addRating(-10);
+			break;
+		}
+		clog(NetMessageSummary) << "GetNodeData (" << dec << count << "entries)";
+
+		// return the requested blocks.
```
Comment (arkpar) [inline libethereum/EthereumPeer.cpp]: forgot to update comment

File: libethereum/EthereumPeer.cpp [hunk 2]
```diff
@@ -474,6 +474,40 @@ bool EthereumPeer::interpret(unsigned _id, RLP const& _r)
...
+		bytes rlp;
+		unsigned n = 0;
+		auto numItemsToSend = std::min(count, c_maxBlocks);
```
Comment (arkpar) [inline libethereum/EthereumPeer.cpp]: better use a new constant

**Comment by chfast** [comment]
@arkpar comments addressed.

File: libethereum/EthereumPeer.cpp [hunk 3]
```diff
@@ -474,6 +474,40 @@ bool EthereumPeer::interpret(unsigned _id, RLP const& _r)
...
+		bytes rlp;
+		unsigned n = 0;
+		auto numItemsToSend = std::min(count, c_maxNodes);
+		for (unsigned i = 0; i < numItemsToSend && rlp.size() < c_maxPayload; ++i)
+		{
+			auto h = _r[i].toHash<h256>();
+			auto node = host()->db().lookup(h);
+			if (!node.empty())
+			{
+				auto nodeData = RLPStream{}.append(node).out();
```
Comment (arkpar) [inline libethereum/EthereumPeer.cpp]: I think node data is already RLP encoded, no need to encode it twice

**Comment by chfast** [comment]
I've removed node data double encoding.
Added reponse to GetReceipts too.

**Pull request merged**